Shop_GXPEngine_base/GXPEngine/Shop/View/InventoryView.cs

```diff
@@ -123,33 +120,48 @@ private void DrawBackground()
 
         private void DrawItems()
         {
-            List<Item> items = inventoryController.Trader.Inventory.GetItems();
-            for (int index = 0; index < items.Count; index++)
+            for (int i = 0; i < inventoryController.Trader.Inventory.ItemCount; i++)
             {
-                Item item = items[index];
-                int iconX = GetColumnByIndex(index) * SPACING + MARGIN;
-                int iconY = GetRowByIndex(index) * SPACING + MARGIN;
+                DrawableComponent drawable = inventoryController.Trader.Inventory.GetItemByIndex(i).GetComponent<DrawableComponent>();
+                if (drawable == null)
+                    continue;
 
-                if (item == inventoryController.Trader.Inventory.GetSelectedItem())
-                    DrawSelectedItem(item, iconX, iconY);
+                int iconX = GetColumnByIndex(i) * SPACING + MARGIN;
+                int iconY = GetRowByIndex(i) * SPACING + MARGIN;
+
+                if (drawable == inventoryController.Trader.Inventory.GetSelectedItem().GetComponent<DrawableComponent>())
```

Comment:
Why not just compare the two items (the one accessed by index and the one selected) instead of comparing the Drawable component? Assuming that components will be unique, that can allow two separate items to add the same component to themselves, and this line potentially cause errors.